android/apollo/src/com/andrew/apollo/provider/RecentSongStore.java

```diff
@@ -0,0 +1,161 @@
+            sInstance = new RecentSongStore(context.getApplicationContext());
+        }
+        return sInstance;
+    }
+
+    public void addSongId(final Long songId, final String songName,
+                          final String artistName, final String albumName,
+                          final long duration) {
+
+        if (songId == null || songName == null || artistName == null || albumName == null ||
+                duration == 0) {
+            return;
+        }
+
+        try {
+            final ContentValues values = new ContentValues(6);
+            values.put(RecentStoreColumns.ID, songId);
+            values.put(RecentStoreColumns.SONG_NAME, songName);
+            values.put(RecentStoreColumns.ARTIST_NAME, artistName);
+            values.put(RecentStoreColumns.ALBUM_NAME, albumName);
+            values.put(RecentStoreColumns.DURATION, duration);
+            values.put(RecentStoreColumns.TIME_PLAYED, System.currentTimeMillis());
+
+            writableDatabase.beginTransaction();
+            writableDatabase.delete(RecentStoreColumns.TABLE_NAME,
+                RecentStoreColumns.ID + " = ?", new String[]{ String.valueOf(songId) });
+            writableDatabase.insert(RecentStoreColumns.TABLE_NAME, null, values);
+            writableDatabase.setTransactionSuccessful();
+            writableDatabase.endTransaction();
+        } catch (Throwable e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public String getSongName(final String key) {
+        if (TextUtils.isEmpty(key)) {
+            return null;
+        }
+
+        final String[] projection = new String[]{
+                RecentStoreColumns.ID,
+                RecentStoreColumns.SONG_NAME,
+                RecentStoreColumns.ARTIST_NAME,
+                RecentStoreColumns.TIME_PLAYED
+        };
+        final String selection = RecentStoreColumns.ARTIST_NAME + "=?";
+        final String[] having = new String[]{ key };
+        Cursor cursor = readableDatabase.query(RecentStoreColumns.TABLE_NAME, projection,
+                selection, having,null, null, RecentStoreColumns.TIME_PLAYED +
+                " DESC", null);
+
+        if (cursor != null && cursor.moveToFirst()) {
+            cursor.moveToFirst();
+            final String song = cursor.getString(
+                    cursor.getColumnIndexOrThrow(RecentStoreColumns.SONG_NAME));
+            cursor.close();
+            return song;
+        }
+
+        if (cursor != null && !cursor.isClosed()) {
+            cursor.close();
+        }
+
+        return null;
+    }
+
+    public void deleteDatabase() {
+        writableDatabase.delete(RecentStoreColumns.TABLE_NAME, null, null);
+    }
+
+    public void removeItem(final long songId) {
+        writableDatabase.delete(RecentStoreColumns.TABLE_NAME,
+                RecentStoreColumns.ID + " = ?",
+                new String[]{String.valueOf(songId)});
+    }
+
+    public interface RecentStoreColumns {
```

Comment:
let's see if we can do away with these fields, and let's reuse the same column name fields defined in `AudioColumns`, we need to reduce the number of symbols used